grr/server/grr_response_server/gui/ui/components/client_details/entry_history_dialog/entry_history_dialog_test.ts

```diff
@@ -57,66 +58,97 @@ describe('Entry History Dialog', () => {
           ],
         })
         .compileComponents();
-
     inject([OverlayContainer], (oc: OverlayContainer) => {
       overlayContainer = oc;
       overlayContainerElement = oc.getContainerElement();
     })();
-  }));
-
-  afterEach(() => {
-    overlayContainer.ngOnDestroy();
-  });
+  }
 
   it('is created successfully', () => {
+    configureTestingModule({
```

Comment:
Is it possible to still call configureTestingModule in beforeEach, but let it inject EntryHistoryDialogParams using `useFactory: () => providedData` and change `providedData` in each test function before `TestBed.createComponent` is called?

If the injection is only executed when TestBed.createComponent is called, this would still allow setting up the TestBed in beforeEach.

Reply:
Trying that resulted in this error. I tried searching for a solution, but wasn't very lucky.
```
Error: Cannot call Promise.then from within a sync test.
    at SyncTestZoneSpec.push../node_modules/zone.js/dist/zone-testing.js.SyncTestZoneSpec.onScheduleTask (http://localhost:9876/_karma_webpack_/node_modules/zone.js/dist/zone-testing.js:368:1)
    at ZoneDelegate.scheduleTask (http://localhost:9876/_karma_webpack_/node_modules/zone.js/dist/zone-evergreen.js:378:1)
    at Zone.scheduleTask (http://localhost:9876/_karma_webpack_/node_modules/zone.js/dist/zone-evergreen.js:210:1)
    at Zone.scheduleMicroTask (http://localhost:9876/_karma_webpack_/node_modules/zone.js/dist/zone-evergreen.js:230:1)
    at scheduleResolveOrReject (http://localhost:9876/_karma_webpack_/node_modules/zone.js/dist/zone-evergreen.js:847:1)
    at ZoneAwarePromise.then (http://localhost:9876/_karma_webpack_/node_modules/zone.js/dist/zone-evergreen.js:979:1)
    at ApplicationInitStatus.runInitializers (http://localhost:9876/_karma_webpack_/node_modules/@angular/core/__ivy_ngcc__/fesm2015/core.js:40425:1)
    at R3TestBedCompiler.finalize (http://localhost:9876/_karma_webpack_/node_modules/@angular/core/__ivy_ngcc__/fesm2015/testing.js:1898:82)
    at TestBedRender3.get testModuleRef [as testModuleRef] (http://localhost:9876/_karma_webpack_/node_modules/@angular/core/__ivy_ngcc__/fesm2015/testing.js:3253:1)
    at TestBedRender3.inject (http://localhost:9876/_karma_webpack_/node_modules/@angular/core/__ivy_ngcc__/fesm2015/testing.js:3110:1)
```

Reply:
Can you post a gist with the code that triggers the error?

Reply:
I managed to find the issue and solve it